File: app/src/main/java/com/softaai/designpatterns/java/structural/composite/Employee.java
package com.softaai.designpatterns.java.structural.composite;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
public class Employee {
    private String name ;
    private String dept;
    private int salary;
    private List<Employee> subordinates;

    public Employee(String name, String dept, int salary){
        this.name = name;
        this.dept = dept;
        this.salary = salary;
        subordinates = new ArrayList<>();
    }

    public void add(Employee e){
        subordinates.add(e);
    }

    public void remove(Employee e){
        subordinates.remove(e);
    }

    public List<Employee> getSubordinates(){
        return subordinates;
    }

    public String toString(){
        return ("Employee : [ name : " + name + ", dept : " + dept + ", salary : " + salary + " ]");
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/composite/Employee.kt
package com.softaai.designpatterns.kotlin.structural.composite


/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
class Employee(name:String, dept:String, salary:Int) {
    private var name:String
    private var dept:String
    private var salary:Int
    private var subordinates : MutableList<Employee>

    init {
        this.name = name
        this.dept = dept
        this.salary = salary
        this.subordinates = ArrayList<Employee>()
    }

    fun add(e : Employee){
        this.subordinates.add(e)
    }

    fun remove(e : Employee){
        this.subordinates.remove(e)
    }

    fun getSubordinates() : MutableList<Employee> {
        return subordinates
    }

    override fun toString() : String{
      return ("Employee :[ Name : " + name + ", dept : " + dept + ", salary : " + salary + " ]")
    }

}